packages/ckeditor5-utils/src/translation-service.ts

```diff
@@ -177,17 +180,17 @@ export function _translate( language: string, message: Message, quantity: number
 		throw new CKEditorError( 'translation-service-quantity-not-a-number', null, { quantity } );
 	}
 
-	const numberOfLanguages = getNumberOfLanguages();
+	const numberOfLanguages = getNumberOfLanguages( translations );
 
 	if ( numberOfLanguages === 1 ) {
 		// Override the language to the only supported one.
 		// This can't be done in the `Locale` class, because the translations comes after the `Locale` class initialization.
-		language = Object.keys( global.window.CKEDITOR_TRANSLATIONS )[ 0 ];
+		language = translations ? Object.keys( translations )[ 0 ] : Object.keys( global.window.CKEDITOR_TRANSLATIONS )[ 0 ];
```

Comment:
Maybe we can simplify this a bit:

```suggestion
		language = Object.keys( translations || global.window.CKEDITOR_TRANSLATIONS )[ 0 ];
```